### DISCUSSION TITLE
What's the use of the autoRemove in the filterFns

### DISCUSSION THREAD

**Discussion by esponges** [KevinVandy/material-react-table#223] [Q&A] [answered]
I was checkint the `filterFns.ts` functions and I was curious of its use in each one. 

E.g. with fuzzy:

```
// src/filterFns.ts
const fuzzy = <TData extends Record<string, any> = {}>(
  row: Row<TData>,
  columnId: string,
  filterValue: string | number,
  addMeta: (item: RankingInfo) => void,
) => {
  const itemRank = rankItem(row.getValue(columnId), filterValue as string, {
    threshold: rankings.MATCHES,
  });
  addMeta(itemRank);
  return itemRank.passed;
};

fuzzy.autoRemove = (val: any) => !val;
```

**Comment by KevinVandy** [comment, marked as answer]
You can find more out about some of these functions here: https://tanstack.com/table/v8/docs/api/features/filters#filterfnautoremove

Pretty much just means if the value is falsy, then the `columnFilters` state entry for that filter will removed from that array.

**Reply by esponges** [reply]
> Pretty much just means if the value is falsy, then the `columnFilters` state entry for that filter will removed from that array.

I would have never understand that from the documentation. Much clearer. Thanks!
